components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/config/APIMConfigService.java

```diff
@@ -138,7 +138,7 @@ public void updateExternalStoreConfig(String organization, String externalStoreC
      * @param organization organization of the user.
      * @throws APIManagementException throw if self-signUp configuration couldn't retrieve.
      */
-    public String getSelfSighupConfig(String organization) throws APIManagementException;
+    public Object getSelfSighupConfig(String organization) throws APIManagementException;
```

Comment:
Please don't return `Object` :D

Reply:
Here, we have to return a UserRegistrationConfigDTO in APIMConfigServiceImpl.getSelfSighupConfig() method. Again we have to implement APIMConfigService.getSelfSighupConfig() interface. But we cannot change the return type of the interface to UserRegistrationConfigDTO as the same interface is implemented by choreo.